### PR TITLE
[Release] Pin upstreams for release 0.18

### PR DESCRIPTION
**Pull request by n3wscott**
Pin upstreams for release 0.18

### CODE REVIEW

**Review by pierDipi — APPROVED** [review]
/lgtm

File: hack/update-deps.sh [hunk 1]
```diff
@@ -32,8 +32,8 @@ cd "${ROOT_DIR}"
 # The list of dependencies that we track at HEAD and periodically
 # float forward in this repository.
 FLOATING_DEPS=(
-  "knative.dev/test-infra@${VERSION}"
-  "knative.dev/eventing@${VERSION}"
+  "knative.dev/test-infra@release-0.18"
+  "knative.dev/eventing@release-0.18"
```
Comment (pierDipi) [inline hack/update-deps.sh:36]: eventing? aren't only pkg and test-infra?

**Comment by n3wscott** [comment]
added in pkg
/unhold

**Review by pierDipi — APPROVED** [review]
/lgtm

**Pull request merged**